pymeasure/instruments/minicircuits/usb_2sp2t_dch.py

```diff
@@ -0,0 +1,101 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument, RangeException
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+import clr
+clr.AddReference('mcl_SolidStateSwitch_NET45')
+from mcl_SolidStateSwitch_NET45 import USB_Digital_Switch
+
+import numpy as np
+import time
+
+
+class USB_2SP2T_DCH():
```

Comment:
To stay within the PyMeasure structure, I would suggest making an `Adapter` class (inheriting the normal `Adapter` class and writing the basic `write`, `ask`, which is also responsible for making the connection) and making this instrument a subclass of `Instrument` (like you normally do with instruments) where you pass an instance of this `Adapter`-subclass to take care of the communication (something like this is done for example for the Attocube instruments, though there it is a subclass of the TelnetAdapter rather than of the base Adapter). That way you retain the abstraction that PyMeasure has and you can use the standard features (e.g. the `setting`, `measurement`, `control` properties) in the class definition.